test/gitea_test.go

```diff
@@ -720,6 +720,110 @@ func verifyProvinance(t *testing.T, topts *tgitea.TestOpts, expectedOutput, cNam
+		yamlFiles[fmt.Sprintf(".tekton/pr%d.yaml", i)] = "testdata/pipelinerun.yaml"
+	}
+	topts := &tgitea.TestOpts{
+		SkipEventsCheck:       true,
+		TargetEvent:           triggertype.PullRequest.String(),
+		NoPullRequestCreation: true,
+		YAMLFiles:             yamlFiles,
+		CheckForNumberStatus:  numPipelines,
+		ConcurrencyLimit:      github.Int(3),
+		ExpectEvents:          false,
+	}
+
+	verifyConcurrency(t, topts, "HELLOMOTO", "step-task")
+}
+
+func verifyConcurrency(t *testing.T, topts *tgitea.TestOpts, expectedOutput, cName string) {
+	ctx := context.Background()
+	topts.ParamsRun, topts.Opts, topts.GiteaCNX, _ = tgitea.Setup(ctx)
+	assert.NilError(t, topts.ParamsRun.Clients.NewClients(ctx, &topts.ParamsRun.Info))
+	topts.TargetRefName = names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pac-e2e-test")
+	topts.TargetNS = topts.TargetRefName
+	ctx, err := cctx.GetControllerCtxInfo(ctx, topts.ParamsRun)
+	assert.NilError(t, err)
+	assert.NilError(t, pacrepo.CreateNS(ctx, topts.TargetNS, topts.ParamsRun))
+	globalNs, _, err := params.GetInstallLocation(ctx, topts.ParamsRun)
+	assert.NilError(t, err)
+	ctx = info.StoreNS(ctx, globalNs)
+
+	err = tgitea.CreateCRD(ctx, topts,
+		v1alpha1.RepositorySpec{
+			ConcurrencyLimit: github.Int(2),
+		},
+		true)
+	assert.NilError(t, err)
+
+	defer (func() {
+		if os.Getenv("TEST_NOCLEANUP") != "true" {
+			topts.ParamsRun.Clients.Log.Infof("Cleaning up global repo %s in %s", info.DefaultGlobalRepoName, globalNs)
+			err = topts.ParamsRun.Clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(globalNs).Delete(
+				context.Background(), info.DefaultGlobalRepoName, metav1.DeleteOptions{})
+			assert.NilError(t, err)
+		}
+	})()
+
+	_, f := tgitea.TestPR(t, topts)
+	defer f()
+	targetRef := topts.TargetRefName
+	entries, err := payload.GetEntries(topts.YAMLFiles, topts.TargetNS, topts.DefaultBranch, topts.TargetEvent, map[string]string{})
+	assert.NilError(t, err)
+	// topts.TargetRefName = topts.DefaultBranch
+
+	scmOpts := &scm.Opts{
+		GitURL:        topts.GitCloneURL,
+		Log:           topts.ParamsRun.Clients.Log,
+		WebURL:        topts.GitHTMLURL,
+		TargetRefName: topts.TargetRefName,
+		BaseRefName:   topts.DefaultBranch,
+	}
+	scm.PushFilesToRefGit(t, scmOpts, entries)
+
+	pr, _, err := topts.GiteaCNX.Client.CreatePullRequest(topts.Opts.Organization, targetRef, gitea.CreatePullRequestOption{
+		Title: "Test Pull Request - " + targetRef,
+		Head:  targetRef,
+		Base:  options.MainBranch,
+	})
+	assert.NilError(t, err)
+	topts.PullRequest = pr
+	topts.ParamsRun.Clients.Log.Infof("PullRequest %s has been created", pr.HTMLURL)
+	topts.CheckForStatus = "success"
+	tgitea.WaitForStatus(t, topts, "heads/"+targetRef, "", false)
+	// check the output of the PipelineRun logs
+	err = twait.RegexpMatchingInPodLog(context.Background(), topts.ParamsRun, topts.TargetNS, "pipelinesascode.tekton.dev/event-type=pull_request", cName, *regexp.MustCompile(expectedOutput), "", 2)
```

Comment:
I do not understand this test. There are 5 pipelineruns defined, right? With the concurrency limit of 2, I expect all 5 of them to run, but only 2 of them concurrently. Eventually you should get here 5 matches, but you only get 2.

Reply:
@enarha this will wait for until all five have been done.

Reply:
Can you elaborate how are you trying to change it? I mean describe in words what's the purpose of this test and how you want to implement it?

Reply:
I want to test that `concurrency_limit` is working while set by global repository and in other test `concurrency_limit`.

Reply:
I still do not think we actually verify the concurrency here. To verify the concurrency we need to verify that no more than 2 (in this case) pipelineruns are running concurrently. I do understand the test as is do check the case where the bug was encountered and fix. Bottom line is I think we can merge that PR as is (given the failing test pass), but we need to ensure we have a test which covers the actual concurrency as well (maybe we do? I'm still not familiar with the code in the repo.).

Reply:
yes @enarha, I agree with you it should check for concurrent PR and it needs to figure out a way that how to do it because at the moment other already written PAC tests do it in the same way I did it.

Reply:
OK. As I mentioned, we can merge that as is and I'm resolving this thread.